app/src/main/java/org/dslul/openboard/inputmethod/latin/suggestions/SuggestionStripView.java

```diff
@@ -291,6 +291,12 @@ private void clear() {
         for (final TextView word : mWordViews) {
             word.setOnTouchListener(null);
         }
+
+        // Make the tool bar and pinned keys visible if setInlineSuggestionsView() was called
+        if (mToolbarExpandKey.getVisibility() != VISIBLE) {
```

Comment:
Checking the current visibility before setting anything is not necessary. A very similar check is also done internally.

Does it work if you set the visibility in `updateKeys`? This would avoid setting the visibility to `VISIBILE` and then setting it to `GONE` when `setInlineSuggestionsView` is called.

Reply:
Yes it works when setting the visibility in `updateKeys`.